ImperatorToCk2/src/main/java/ImperatorToCK2/Processing.java

```diff
@@ -1,707 +1,736 @@
+                }
+
+                try {
+
+                    qaaa = qaaa.split("=")[0];
+                    number = Integer.parseInt(qaaa);    
+
+                    provList[number] = region;   
+
+                }catch (java.lang.NumberFormatException exception) {
+                }
+                aqq = aqq + 1;    
+            }
+        }catch (java.util.NoSuchElementException exception){
+            endOrNot = false;
+
+        }   
+
+        return provList;
+
+    }
+
+    public static String randomizeColor ()
+    {
+
+        int flag = 0;
+        int aqq = 0;
+        String qaaa = "aa";
+        int Rng1 = (int) (Math.random() * 255);
+        int Rng2 = (int) (Math.random() * 255);
+        int Rng3 = (int) (Math.random() * 255);
+        String color = Integer.toString(Rng1) + " " + Integer.toString(Rng2) + " " + Integer.toString(Rng3); 
+
+        return color;
+
+    }
+
+    public static String[] convertProvConvList (String name, String outputDest) throws IOException //Converts standard mapper tool format into regular format
+    {
+
+        String tab = "	";
+
+        FileInputStream fileIn= new FileInputStream(name);
+        Scanner scnr= new Scanner(fileIn);
+
+        FileOutputStream fileOut= new FileOutputStream(outputDest);
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int flag = 0;
+        boolean endOrNot = true;
+        String qaaa;
+        String[] output;   // Owner Culture Religeon PopTotal Buildings
+        output = new String[15000];
+
+        output[0] = "peq"; //default for no owner, uncolonized province
+        output[1] = "99999"; //default for no culture, uncolonized province with 0 pops
+
+        char bracket1 = 123;
+        char bracket2 = 125;
+
+        String bracket1Str = bracket1+" ";
+
+        bracket1Str = bracket1Str.replace(" ","");
+
+        String bracket2Str = bracket2+" ";
+
+        bracket2Str = bracket2Str.replace(" ","");
+
+        int aq2 = 0;
+
+        try {
+            while (endOrNot = true){
+
+                qaaa = scnr.nextLine();
+
+                if (qaaa.split("ink = ")[0].equals(tab+"l")){
+                    qaaa = qaaa.split("ink = ")[1];
+                    qaaa = qaaa.split(tab)[0];
+                    qaaa = qaaa.substring(2,qaaa.length()-2);;
+                    qaaa = qaaa.replace(" = ",",");
+
+                    String[] provinceMappings = qaaa.split(" ");
+
+                    String impProv = "a";
+
+                    int aqq = 0;
+
+                    while (aqq < provinceMappings.length) {
+
+                        if (provinceMappings[aqq].split(",")[0].equals("ck2")){
+                            impProv = provinceMappings[aqq].split(",")[1];
+                            aqq = 99999;
+
+                        }
+                        else {aqq = aqq + 1;}  
+
+                    }
+                    aqq = 0;   
+                    if (impProv != "a") {
```

Comment:
My Java knowledge is poor, but wouldn't

```suggestion
                    if (!"a".equals(impProv)) {
```

for string inequality checks be better?

Reply:
I'm not sure, but `impProv != "a"` works and doesn't cause any errors. If it works, it works